iothub/service/src/IotHubServiceClientOptions.cs

```diff
@@ -59,6 +61,23 @@ public IotHubServiceClientOptions()
         /// </summary>
         public HttpClient HttpClient { get; set; }
 
+        /// <summary>
+        /// Whether to use web sockets or not.
+        /// </summary>
+        /// <remarks>
+        /// Only used for AMQP or MQTT.
+        /// </remarks>
+        public bool UseWebSocketOnly { get; set; }
+
+        /// <summary>
+        /// Service client transport settings used for AMQP/MQTT operations.
+        /// </summary>
+        /// <remarks>
+        /// If provided, all other settings will be ignored, otherwise,
+        /// a transport settings client will be created based on other provided settings.
+        /// </remarks>
+        public ServiceClientTransportSettings TransportSettings { get; set; }
```

Comment:
This class doesn't appear to add any options that we didn't already have. Why are you adding it here?

Reply:
This ```IotHubServiceClientOptions``` class already allows users to provide a proxy that the HTTP client and AMQP client can use. The only other option is ```ConnectionLeaseTimeoutMilliseconds``` which we are intentionally no longer exposing. Users who want to configure settings at that level of granularity are expected to pass in an HTTP client they themselves constructed